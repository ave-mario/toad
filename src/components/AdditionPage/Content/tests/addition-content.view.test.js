import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import AdditionContent from '../addition-content.view';
import additions from './mock-store';

describe('Addition Content view', () => {
  const load = () => {};

  // test('should open modal view after click on Edit button', () => {
  //   const mockStore = configureStore();
  //   const store = mockStore({});
  //   const wrapper = mount(
  //     <Provider store={store}>
  //       <AdditionContent {...additions} load={load} />
  //     </Provider>
  //   );
  //   wrapper
  //     .find(`button`)
  //     .at(0)
  //     .simulate('click');
  //   wrapper.update();

  //   expect(wrapper.state('isShowModal')).toEqual(true);
  // });

  test('should must have buttons twice the number of services', () => {
    const component = mount(<AdditionContent {...additions} load={load} />);
    let buttons = component.find(`button`);
    expect(buttons.length).toBe(additions.docs.length * 2);
  });

  test('should render addition table', () => {
    const component = shallow(<AdditionContent {...additions} load={load} />);
    expect(component).toMatchSnapshot();
  });
});
