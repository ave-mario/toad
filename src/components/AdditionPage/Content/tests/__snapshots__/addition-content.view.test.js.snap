// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Addition Content view should render addition table 1`] = `
<AdditionTable
  docs={
    Array [
      Object {
        "_id": "5d024a6ced7a017a7c437b44",
        "name": "Service fisrtyjgjyssdhs",
        "price": 2.8,
      },
      Object {
        "_id": "5d024a6ced7a017a7c437b45",
        "name": "Fridhgnge",
        "price": 10,
      },
      Object {
        "_id": "5d08bc215fed473525340517",
        "name": "Rooms",
        "price": 0.6,
      },
      Object {
        "_id": "5d08f91b623ca1aaecb1fff5",
        "name": "Second service",
        "price": 1.6,
      },
      Object {
        "_id": "5d0b55b2c0e3612e2dec279d",
        "name": "Roomshj,vhj",
        "price": 2,
      },
      Object {
        "_id": "5d0b55cfc0e3612e2dec279e",
        "name": "Roomsfgncgh",
        "price": 1.5,
      },
      Object {
        "_id": "5d0c9219034c70c2e09b6a6f",
        "name": "kghjmn",
        "price": 0,
      },
    ]
  }
  i18n={Object {}}
  limit={10}
  load={[Function]}
  page={1}
  pages={1}
  t={[Function]}
  tReady={true}
  total={8}
/>
`;
